app/helpers/merge_requests_helper.rb

```diff
@@ -108,6 +108,72 @@ def total_users_after_merge_text(merge_request)
     "#{"#{count} user".pluralize(count)} after merge"
   end
 
+  def total_stock_owners_after_merge_text(merge_request)
+    count = merge_request.total_stock_owners_after_merge
+    "#{"#{count} stock owner".pluralize(count)} after merge"
+  end
+
+  def total_managing_agents_after_merge_text(merge_request)
+    count = merge_request.total_managing_agents_after_merge
+    "#{"#{count} managing agent".pluralize(count)} after merge"
+  end
+
+  def related_organisations(merge_request, relationship_type)
+    organisations =  merge_request.absorbing_organisation.send(relationship_type.pluralize).visible + merge_request.merging_organisations.flat_map { |org| org.send(relationship_type.pluralize).visible }
+    organisations += [merge_request.absorbing_organisation] + merge_request.merging_organisations
```

Comment:
The related organisations returns all organisations including the absorbing org itself and the merging orgs themselves. To account for one or the other being a stock owner or managing agents of the other.